File: xsListContexts.c
/*
 Copyright 2012-2014 Scianta Analytics LLC   All Rights Reserved.  
 Reproduction or unauthorized use is prohibited. Unauthorized
 use is illegal. Violators will be prosecuted. This software 
 contains proprietary trade and business secrets.            

 Program: xsListContexts

 Usage: xsListContexts [-s scope]
    -s the scope containing the contexts to list (defaults to global)

 Description:
    List the contexts in a scope (private, app or global)
*/
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include "saLicensing.h"
#include "saSplunk.h"

extern char *optarg;
extern int optind, optopt;

extern bool saSplunkGetContextPath(char *, char *, int, char *, char *);
extern char *saSplunkGetRoot(char *);
extern int saSplunkGetScope(char *);
extern saSplunkInfoPtr saSplunkLoadHeader();
extern bool saSplunkReadInfoPathFile(saSplunkInfoPtr);
extern void saListDir(char *, char *, bool, FILE *, char *);

int main(int argc, char *argv[])
{
    bool argError;
    int c;
    int scope = SA_SPLUNK_SCOPE_GLOBAL;
    
    if (!isLicensed())
        exit(EXIT_FAILURE);

    while ((c = getopt(argc, argv, "s:")) != -1) 
    {
        switch(c)
        {
            case 's':
                scope = saSplunkGetScope(optarg);
                break;
            case '?':
                fprintf(stderr, "xsListContexts-F-101: Unrecognised option: -%c\n", optopt);
                argError = true;
        }
    }
    if (argError)
    {
        fprintf(stderr, "xsListContexts-F-103: Usage: xsListContexts [-s scope]");
        exit(EXIT_FAILURE);
    }

    saSplunkInfoPtr p = saSplunkLoadHeader();
    if (p == NULL)
    {
        fprintf(stderr, "xsListContexts-F-105: Can't get Header record\n");
        exit(EXIT_FAILURE);
    }

    if (saSplunkReadInfoPathFile(p) == false)
    {
        fprintf(stderr, "xsListContexts-F-107: Can't read search results file %s\n",
                p->infoPath == NULL ? "NULL" : p->infoPath);
        exit(EXIT_FAILURE);
    }

    char path[1024];
    char *root = saSplunkGetRoot(argv[0]);
    if (saSplunkGetContextPath(path, root, scope, p->app, p->user) == false)
    {
        fprintf(stderr, "xsListContexts-F-109: failed to get context path\n");
        exit(EXIT_FAILURE);
    }
    saListDir(path, ".context", true, stdout, "context");
}

File: python/xsListContexts.py
#
# Copyright 2012-2014 Scianta Analytics LLC   All Rights Reserved.  
# Reproduction or unauthorized use is prohibited. Unauthorized
# use is illegal. Violators will be prosecuted. This software 
# contains proprietary trade and business secrets.            
#
import sys, subprocess, os, platform, time
import splunk.Intersplunk as si

if __name__ == '__main__':

    scope = '3'
    try:
        if len(sys.argv) < 1:
            raise Exception("xsListContexts-F-001: Usage: xsListContexts [scope]") 

        if len(sys.argv) == 2:
            scope = sys.argv[1]
            scope = scope.lower()

        binary = os.path.dirname(sys.argv[0]) + "/" +  platform.system() + "/" + platform.architecture()[0] + "/xsListContexts"
	if (platform.system() == 'Windows'):
            binary = binary + ".exe"
        if not os.path.isfile(binary):
            raise Exception("xsListContexts-F-000: Can't find binary file " + binary)

        subprocess.call([binary, '-s', scope])

        if platform.system() == 'Windows':
            sys.stdout.flush()
            time.sleep(1.0)

    except Exception, e:
        si.generateErrorResults(e)
